### PR TITLE
Signals: Common legitimate mail services

### PR DESCRIPTION
**Pull request by bfilar**

### CODE REVIEW

File: signals/sender/sender_legitimate_domain_contain_recipient_addr.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+name: "Sender: Legitimate domain where recipient in sender's email address local part"
```
Comment (aidenmitchell) [inline signals/sender/sender_legitimate_domain_contain_recipient_addr.yml:1]: Should we be checking for the "recipient in sender's email address local part"?

File: signals/sender/sender_legitimate_domain_negation.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: "Sender: Legitimate message senders"
```
Comment (aidenmitchell) [inline signals/sender/sender_legitimate_domain_negation.yml:1]: What's the idea here, to fire on secure mailers? Right now, it looks like this is just "not a known secure mailer".

**Review by aidenmitchell — CHANGES_REQUESTED** [review]

File: signals/sender/sender_legitimate_domain_negation.yml (hunk 2 above)
Comment (bfilar) [reply signals/sender/sender_legitimate_domain_negation.yml:1]: yeah. i need to `or` these and remove the `not` logic

Comment (aidenmitchell) [inline signals/sender/sender_legitimate_domain_negation.yml:1]:
```suggestion
name: "Sender: Legitimate secure message senders"
```

File: signals/sender/sender_legitimate_domain_negation.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+type: "query"
+source: |  
+  (
+    sender.email.domain.root_domain in ("protectedtrust.com", "medallia.com", "icims.com", "workday.com")
+    or any(body.links,
+            .href_url.domain.root_domain == sender.email.domain.root_domain
+    )
```
Comment (aidenmitchell) [inline signals/sender/sender_legitimate_domain_negation.yml:8]:
```suggestion
```

I'm not sure these senders _always_ send secure messages... thoughts?
as for L6-8, it's probably a remnant from pulling these signals out

File: signals/sender/sender_legitimate_ups_domain.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+name: "Sender: Legitimate UPS domain"
+type: "query"
+source: |  
+  sender.email.domain.root_domain in ("ups.com", "upsemail.com")
```
Comment (aidenmitchell) [inline signals/sender/sender_legitimate_ups_domain.yml:4]: May want to add a DMARC check here, unless risk score will weigh a DMARC failure more/a failure would be accounted for elsewhere.

File: signals/sender/sender_is_known_msft_domain copy.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+name: "Sender: Known MSFT domain"
+type: "query"
+source: |
```
Comment (aidenmitchell) [inline signals/sender/sender_is_known_msft_domain copy.yml:3]: May want to add a DMARC check here, unless risk score will weigh a DMARC failure more/a failure would be accounted for elsewhere.

File: signals/sender/sender_is_known_boa_domain.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+name: "Sender: Known Bank of America domain"
+type: "query"
+source: |
```
Comment (aidenmitchell) [inline signals/sender/sender_is_known_boa_domain.yml:3]: May want to add a DMARC check here, unless risk score will weigh a DMARC failure more/a failure would be accounted for elsewhere.

File: signals/sender/sender_is_known_fax_service.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+name: "Sender: Known fax mailers"
+type: "query"
+source: |
```
Comment (aidenmitchell) [inline signals/sender/sender_is_known_fax_service.yml:3]: May want to add a DMARC check here, unless risk score will weigh a DMARC failure more/a failure would be accounted for elsewhere.

**Review by aidenmitchell — CHANGES_REQUESTED** [review]

File: signals/sender/sender_legitimate_ups_domain.yml (hunk 4 above)
Comment (bfilar) [reply signals/sender/sender_legitimate_ups_domain.yml:4]: Yeah we have signals already for DMARC failures

**Review by aidenmitchell — APPROVED** [review]

**Pull request merged**